rich/traceback.py

```diff
@@ -367,6 +367,8 @@ def safe_str(_object: Any) -> str:
                 if filename and not filename.startswith("<"):
                     if not os.path.isabs(filename):
                         filename = os.path.join(_IMPORT_CWD, filename)
+                if frame_summary.f_locals.get("_rich_traceback_omit", False) is True:
```

Comment:
The `_rich_traceback_guard` mechanism is triggered by just the presence of the local (i.e. doesn't care about the value). I think this should be consistent.

On reflection I think we should look at the value to make it easier to add a condition. The default should be `False` if the local isn't defined.

Something like this:

```python
if getattr(frame_summary.f_locals, "_rich_traceback_omit", False):
   ...
```

The `"_rich_traceback_guard"` should also be updated in the same way.

Reply:
I see! However, `f_locals` being a Python dictionary it doesn't support the `getattr` protocol if I'm not wrong?

But I updated the `_rich_traceback_guard` management so that both look the same. Also, they are both activated as soon as the value is defined in the locals and its value is _true-ish_ value, rather than strict `True` :slightly_smiling_face: 
--> done there: [f0ca6ff](https://github.com/Textualize/rich/pull/2226/commits/f0ca6ff2bc0dbca2ffcd2a5114d04b92679ecbb3)

Reply:
My bad, should of course be `.get`.